src/escape.rs

```diff
@@ -0,0 +1,81 @@
+//! Escape characters that may have special meaning in a shell, including spaces.
+//! This is a modified version of the [`shell-escape::unix`] module of [`shell-escape`] crate.
+//! 
+//! [`shell-escape`]: https://crates.io/crates/shell-escape
+//! [`shell-escape::unix`]: https://docs.rs/shell-escape/latest/src/shell_escape/lib.rs.html#101
+
+
+fn whitelisted(ch: char) -> bool {
+    match ch {
+        'a'..='z' | 'A'..='Z' | '0'..='9' | '-' | '_' | '=' | '/' | ',' | '.' | '+' => true,
+        _ => false,
+    }
+}
+
+/// Escape characters that may have special meaning in a shell, including spaces.
+/// 
+/// **Note**: This function is an adaptation of [`shell-escape::unix::escape`].
+/// This function exists only for type compatibility and the implementation is
+/// almost exactly the same as [`shell-escape::unix::escape`].
+/// 
+/// [`shell-escape::unix::escape`]: https://docs.rs/shell-escape/latest/src/shell_escape/lib.rs.html#101
+/// 
+pub fn escape(s: &[u8]) -> String {
+    let all_whitelisted = s.iter().all(|x| whitelisted(*x as char));
+
+    if !s.is_empty() && all_whitelisted {
+        // All bytes are whitelisted and valid single-byte UTF-8, 
+        // so we can build the original string and return as is.
+        return String::from_utf8(s.to_vec()).unwrap();
+    }
+
+    let mut escaped = String::with_capacity(s.len() + 2);
```

Comment:
Same here, should use `Vec` and then use [`std::os::unix::ffi::OsStringExt::from_vec`](https://doc.rust-lang.org/std/os/unix/ffi/trait.OsStringExt.html#tymethod.from_vec) to convert it to `OsStr`.